src/math/trig.c

```diff
@@ -94,9 +114,113 @@ void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t
 		xtmp = *xn >> (b_idx + 1);
 		ytmp = *b_yn >> (b_idx + 1);
 	}
-	/* Q2.30 format */
+	/* Q2.30 format -sine,cosine*/
 	*th_cdc_fxp = th_rad_fxp;
+}
+
+/**
+ * Arguments	: int32_t realvalue
+ *		  int16_t numiters
+ *		  const int32_t LUT[CORDIC_30B_ITABLE_SIZE][30]
+ * Return Type	: int32_t
+ */
+int32_t iscalarcordic(int32_t realvalue, int16_t numiters,
+		      const int32_t LUT[CORDIC_30B_ITABLE_SIZE], cordic_cfg type)
+{
+	int32_t b_sincosvalue;
+	int32_t b_i;
+	int32_t i;
+	int32_t x = 0;
+	int32_t xdshift;
+	int32_t xshift;
+	int32_t y = 0;
+	int32_t ydshift;
+	int32_t yshift;
+	int32_t z = 0;
+	int16_t j;
+	int16_t k;
+
+	b_sincosvalue = (realvalue >> 1U);
+	if (type == EN_32B_CORDIC_ASINE) {
+	/*ternary operator*/
+		if (b_sincosvalue > cord_arcsincos_q28fl) {
+			x = 0;
+			y = cord_arcsincos_q30fl;
+			z = 843314856;
+		} else {
+			x = cord_arcsincos_q30fl;
+			y = 0;
+			z = 0;
+		}
+	}
+	if  (type == EN_32B_CORDIC_ACOSINE) {
+	/*ternary operator*/
+		if (b_sincosvalue < cord_arcsincos_q28fl) {
+			x = 0;
+			y = cord_arcsincos_q30fl;
+			z = 843314856;
+		} else {
+			x = cord_arcsincos_q30fl;
+			y = 0;
+			z = 0;
+		}
+	}
+	b_sincosvalue <<= 1U;
+	/*DCORDIC(Double CORDIC) algorithm*/
+	i = (int32_t)((int16_t)(numiters - 1));
+	for (b_i = 0; b_i < i; b_i++) {
+		j = (int16_t)((b_i + 1) << 1U);
+		if (j >= 31)
+			j = 31;
+
+		if (b_i < 31)
+			k = (int16_t)b_i;
+		else
+			k = 31;
+
+		xshift = (x >>	(k));
+		xdshift = (x >> (j));
+		yshift = (y >>	(k));
+		ydshift = (y >> (j));
+
+		/*Adjust input with coefficients*/
+		if (type == EN_32B_CORDIC_ASINE) {
```

Comment:
looking at this code, it seems very similar the two blocks. Would it not be simpler to combine the two with "sign" variables? e.g. instead of
```
if asine
...
x = (x - xdshift) + yshift;
...
else if acos
...
x = (x - xdshift) - yshift;
...
```

instead we have something like this (note sign is passed in as mentioned in previous comment regarding config)

```
x = (x - xdshift) + (config.sign * yshift);
```

assuming the multiply is fast enough, may need something else there

Reply:
or alternative would be to ternaries use ternaries to avoid multiplication @ShriramShastry ping on this
`x = x - xdshift + (config.sign ? yshift : -yshift)`

You could also use a macro to do this
`#define NEGATE(is_negative, x) ((is_negative) ? -(x) : (x))`

Reply:
Multiply is fastest on DSP, I think it's better than ternary to invert. With intrinsics definitely.

Reply:
Thanks @cujomalainey @singalsu . Very good suggestion towards optimization.